# **Competição de Data Science - Fase 01**

> **Nome:** Vicenzo Carlim de Sousa


> *Case:* Análise dos dados provenientes das redes sociais da IME Jr.





In [231]:
# Importação de biblioteca necessária

import pandas as pd

# Importação da base de dados

df = pd.read_csv("https://raw.githubusercontent.com/vicenzocarlim/fase01/main/fase_01.csv")

In [232]:
# Exibir as cinco primeiras linhas do DataFrame
df.head()

Nome do Post        Data  \
0             Amostragem_Imagem_Feed_LinkedIn_150721  15/07/2021   
1       AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
2    BingoDoEstudante_Vídeo_Stories_Instagram_110821  11/08/2021   
3     BoaProvaFuvest_Imagem_Stories_Instagram_111221  11/12/2021   
4  CuriosidadesArcane(1)_Imagem_Stories_Instagram...  03/12/2021   

                Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
0             Amostragem           Feed   LinkedIn  Imagem       NaN      NaN   
1           AulõesFuvest        Stories  Instagram  Imagem       NaN    116.0   
2       BingoDoEstudante        Stories  Instagram   Vídeo       NaN     93.0   
3         BoaProvaFuvest        Stories  Instagram  Imagem       NaN     75.0   
4  CuriosidadesArcane(1)        Stories  Instagram  Imagem       NaN     84.0   

   Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
0         NaN    0.0            108.0         0.0                 NaN   
1       119.0    NaN              NaN         NaN                 NaN   
2         NaN    0.0              0.0         0.0                 0.0   
3        75.0    NaN              NaN         NaN                 NaN   
4        88.0    NaN              NaN         NaN                 NaN   

   Comentários  Reações  Compartilhamentos  
0            0     15.0                0.0  
1            0      NaN                1.0  
2            0      0.0                0.0  
3            0      NaN                0.0  
4            1      NaN                1.0


# 1. Tratamento dos dados

**QUESTÃO 01:** Trate os dados.

**1.** Tratar os valores NaN's

In [233]:
# Conferir se existem valores NaN em cada coluna
df.isnull().sum()

Nome do Post            0
Data                    0
Criativo                0
Posicionamento          0
Plataforma              0
Formato                 0
Conteúdo              230
Alcance                23
Impressões              6
Saves                  40
Cliques em Link        56
3-Sec-View             49
Visualizações 100%     84
Comentários             0
Reações                40
Compartilhamentos       5
dtype: int64

Pode-se observar que todas as colunas que possuem valores NaN referem-se a métricas/números/quantidades. Portanto, substituiremos esses valores por 0. 

In [234]:
# Trocar NaN's por 0
df.fillna(0, inplace = True)

**2.** Tratar a coluna 2, que corresponde à variável Data

In [235]:
# Verificar se há alguma data destoante das demais com base no ano
# Para tal, colocar as datas no tipo 'datetime'

import datetime

df["Data"] = pd.to_datetime(df.Data)
# Exibir as 5 primeiras linhas para verificar a troca do formato da data
df.head() 

Nome do Post       Data  \
0             Amostragem_Imagem_Feed_LinkedIn_150721 2021-07-15   
1       AulõesFuvest_Imagem_Stories_Instagram_201221 2021-12-20   
2    BingoDoEstudante_Vídeo_Stories_Instagram_110821 2021-11-08   
3     BoaProvaFuvest_Imagem_Stories_Instagram_111221 2021-11-12   
4  CuriosidadesArcane(1)_Imagem_Stories_Instagram... 2021-03-12   

                Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
0             Amostragem           Feed   LinkedIn  Imagem       0.0      0.0   
1           AulõesFuvest        Stories  Instagram  Imagem       0.0    116.0   
2       BingoDoEstudante        Stories  Instagram   Vídeo       0.0     93.0   
3         BoaProvaFuvest        Stories  Instagram  Imagem       0.0     75.0   
4  CuriosidadesArcane(1)        Stories  Instagram  Imagem       0.0     84.0   

   Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
0         0.0    0.0            108.0         0.0                 0.0   
1       119.0    0.0              0.0         0.0                 0.0   
2         0.0    0.0              0.0         0.0                 0.0   
3        75.0    0.0              0.0         0.0                 0.0   
4        88.0    0.0              0.0         0.0                 0.0   

   Comentários  Reações  Compartilhamentos  
0            0     15.0                0.0  
1            0      0.0                1.0  
2            0      0.0                0.0  
3            0      0.0                0.0  
4            1      0.0                1.0

In [236]:
# Exibir todos os anos que as postagens contemplam

df_serie = pd.Series(df["Data"])
df_serie_rem = df_serie.dt.year.sort_values().drop_duplicates()
df_serie_rem

181    2017
0      2021
10     2022
165    2023
69     2024
199    2025
Name: Data, dtype: int64

In [242]:
# Exibir as datas cujos anos não correspondem à realidade

df_2017 = df.loc[df["Data"].dt.year == 2017]
print(df_2017["Nome do Post"])
print(df_2017["Data"])
df_2023 = df.loc[df["Data"].dt.year == 2023]
print(df_2023["Nome do Post"])
print(df_2023["Data"])
df_2024 = df.loc[df["Data"].dt.year == 2024]
print(df_2024["Nome do Post"])
print(df_2024["Data"])
df_2025 = df.loc[df["Data"].dt.year == 2025]
print(df_2025["Nome do Post"])
print(df_2025["Data"])

140          DiaDoVoluntário_Imagem_Feed_LinkedIn_280821
181    MotivosParaNãoSeInscrever_Imagem_Feed_Instagra...
Name: Nome do Post, dtype: object
140   2017-08-28
181   2017-08-11
Name: Data, dtype: datetime64[ns]
165    FraseSofiaKovalevskaya_Imagem_Feed_Instagram_0...
Name: Nome do Post, dtype: object
165   2023-08-03
Name: Data, dtype: datetime64[ns]
19     DadosPopulação(6)_Vídeo_Stories_Instagram_170721
69    DesafiosMatemáticos(3)_Imagem_Stories_Instagra...
Name: Nome do Post, dtype: object
19   2024-07-17
69   2024-05-07
Name: Data, dtype: datetime64[ns]
199    ReelsAlissa_Vídeo_Reels_Instagram_211121
Name: Nome do Post, dtype: object
199   2025-03-19
Name: Data, dtype: datetime64[ns]


In [243]:
# Trocar os anos incorretos/destoantes pela data correta
# de acordo com o nome do post

df = df.replace(df["Data"][140], "2021-08-28")
df = df.replace(df["Data"][181], "2021-09-13")
df = df.replace(df["Data"][165], "2021-10-06")
df = df.replace(df["Data"][19], "2021-07-17")
df = df.replace(df["Data"][69], "2021-07-05")
df = df.replace(df["Data"][199], "2021-11-21")

# Exibir todos os anos que as postagens contemplam, a partir de agora

df_serie_2 = pd.Series(df["Data"])
df_serie_rem_2 = df_serie_2.dt.year.sort_values().drop_duplicates()
df_serie_rem_2


0     2021
45    2022
Name: Data, dtype: int64

In [244]:
# Voltar a data para o formato DD/MM/AAAA
df["Data"] = pd.to_datetime(df.Data)
df["Data"] = df["Data"].dt.strftime("%d/%m/%Y")
# Exibir as 5 primeiras linhas para verificar a troca do formato da data
df.head()

Nome do Post        Data  \
0             Amostragem_Imagem_Feed_LinkedIn_150721  15/07/2021   
1       AulõesFuvest_Imagem_Stories_Instagram_201221  20/12/2021   
2    BingoDoEstudante_Vídeo_Stories_Instagram_110821  08/11/2021   
3     BoaProvaFuvest_Imagem_Stories_Instagram_111221  12/11/2021   
4  CuriosidadesArcane(1)_Imagem_Stories_Instagram...  12/03/2021   

                Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
0             Amostragem           Feed   LinkedIn  Imagem       0.0      0.0   
1           AulõesFuvest        Stories  Instagram  Imagem       0.0    116.0   
2       BingoDoEstudante        Stories  Instagram   Vídeo       0.0     93.0   
3         BoaProvaFuvest        Stories  Instagram  Imagem       0.0     75.0   
4  CuriosidadesArcane(1)        Stories  Instagram  Imagem       0.0     84.0   

   Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
0         0.0    0.0            108.0         0.0                 0.0   
1       119.0    0.0              0.0         0.0                 0.0   
2         0.0    0.0              0.0         0.0                 0.0   
3        75.0    0.0              0.0         0.0                 0.0   
4        88.0    0.0              0.0         0.0                 0.0   

   Comentários  Reações  Compartilhamentos  
0            0     15.0                0.0  
1            0      0.0                1.0  
2            0      0.0                0.0  
3            0      0.0                0.0  
4            1      0.0                1.0

**3.** Tratar a coluna 4, que corresponde à variável Posicionamento

In [245]:
# Descobrir quais são os tipos de posicionamento possíveis
# para, posteriormente, normalizá-los

df_serie = pd.Series(df["Posicionamento"])
df_serie_rem = df_serie.sort_values().drop_duplicates()
df_serie_rem

215        Fed
0         Feed
200      Reels
118     Storie
24     Stories
156      Story
208       feed
45     stories
Name: Posicionamento, dtype: object

In [246]:
# Normalizar os termos que correspondem aos tipos de posicionamentos

df = df.replace("Fed", "Feed")
df = df.replace("feed", "Feed")
df = df.replace("Storie", "Stories")
df = df.replace("Story", "Stories")
df = df.replace("stories", "Stories")

# Conferir se, de fato, houve a normalização dos termos

df.groupby("Posicionamento").sum()


Conteúdo  Alcance  Impressões  Saves  Cliques em Link  \
Posicionamento                                                          
Feed                 0.0  20933.0     33229.0  240.0          45522.0   
Reels                0.0   5124.0      5503.0   26.0              0.0   
Stories              0.0  22659.0     16736.0    0.0             20.0   

                3-Sec-View  Visualizações 100%  Comentários   Reações  \
Posicionamento                                                          
Feed                   0.0                 0.0          651    4694.0   
Reels                  0.0                 0.0          115     234.0   
Stories                0.0                 0.0           29  100014.0   

                Compartilhamentos  
Posicionamento                     
Feed                        464.0  
Reels                        14.0  
Stories                       7.0

**4.** Tratar a coluna 5, que corresponde à variável plataforma

In [247]:
# Descobrir quais são os tipos de plataforma possíveis
# para, posteriormente, normalizá-los

df_serie = pd.Series(df["Plataforma"])
df_serie_rem = df_serie.sort_values().drop_duplicates()
df_serie_rem

114    Instagram
216     LinkedIn
134     Linkedin
131           ig
Name: Plataforma, dtype: object

In [248]:
# Normalizar os termos que correspondem aos tipos de plataformas

df = df.replace("Linkedin", "LinkedIn")
df = df.replace("ig", "Instagram")

# Conferir se, de fato, houve a normalização dos termos

df.groupby("Plataforma").sum()

Conteúdo  Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
Plataforma                                                                      
Instagram        0.0  48716.0     48985.0  266.0          44605.0         0.0   
LinkedIn         0.0      0.0      6483.0    0.0            937.0         0.0   

            Visualizações 100%  Comentários   Reações  Compartilhamentos  
Plataforma                                                                
Instagram                  0.0          786  102759.0              478.0  
LinkedIn                   0.0            9    2183.0                7.0

**5.** Tratar a coluna 6, que corresponde à variável Formato

In [249]:
# Descobrir quais são os tipos de formato possíveis
# para, posteriormente, normalizá-los

df_serie = pd.Series(df["Formato"])
df_serie_rem = df_serie.sort_values().drop_duplicates()
df_serie_rem

0      Imagem
153     Photo
192     Video
201     Vídeo
79     imagem
108     vídeo
Name: Formato, dtype: object

In [250]:
# Normalizar os termos que correspondem aos tipos de formato

df = df.replace("Photo", "Imagem")
df = df.replace("Video", "Vídeo")
df = df.replace("imagem", "Imagem")
df = df.replace("vídeo", "Vídeo")

# Conferir se, de fato, houve a normalização dos termos

df.groupby("Formato").sum()

Conteúdo  Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
Formato                                                                      
Imagem        0.0  33077.0     46074.0  247.0          45542.0         0.0   
Vídeo         0.0  15639.0      9394.0   19.0              0.0         0.0   

         Visualizações 100%  Comentários   Reações  Compartilhamentos  
Formato                                                                
Imagem                  0.0          746    4785.0              471.0  
Vídeo                   0.0           49  100157.0               14.0

# 2. Análise dos dados

**QUESTÃO 02:** Qual o post mais reagido?


> **Resposta:** O post mais reagido é DiaDaPizza(2)_Vídeo_Stories_Instagram_100721, com 99999 *likes*.



In [251]:
# Procurar o post com a maior quantidade de Reações

mais_reag = df.sort_values("Reações").tail(1)
mais_reag

Nome do Post        Data       Criativo  \
109  DiaDaPizza(2)_Vídeo_Stories_Instagram_100721  07/10/2021  DiaDaPizza(2)   

    Posicionamento Plataforma Formato  Conteúdo  Alcance  Impressões  Saves  \
109        Stories  Instagram   Vídeo       0.0     70.0        70.0    0.0   

     Cliques em Link  3-Sec-View  Visualizações 100%  Comentários  Reações  \
109              0.0         0.0                 0.0            0  99999.0   

     Compartilhamentos  
109                0.0

**QUESTÃO 03:** Qual o post mais comentado?


> **Resposta:** O post mais comentado é SorteioValePresente_Imagem_Feed_Instagram_291021, com 369 comentários.



In [252]:
# Procurar o post com a maior quantidade de Comentários 

mais_com = df.sort_values("Comentários").tail(1)
mais_com

Nome do Post        Data  \
217  SorteioValePresente_Imagem_Feed_Instagram_291021  29/10/2021   

                Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
217  SorteioValePresente           Feed  Instagram  Imagem       0.0    530.0   

     Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
217       618.0   13.0              0.0         0.0                 0.0   

     Comentários  Reações  Compartilhamentos  
217          369     94.0               11.0

**QUESTÃO 04:** Qual o post mais compartilhado?


> **Resposta:** O post mais compartilhado é SemanaPreFuvest_Imagem_Feed_Instagram_060121, com 200 compartilhamentos.



In [253]:
# Procurar o post com o maior número de compartilhamentos

mais_comp = df.sort_values("Compartilhamentos").tail(1)
mais_comp

Nome do Post        Data  \
214  SemanaPreFuvest_Imagem_Feed_Instagram_060121  01/06/2021   

            Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
214  SemanaPreFuvest           Feed  Instagram  Imagem       0.0   6058.0   

     Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
214      8849.0   81.0              0.0         0.0                 0.0   

     Comentários  Reações  Compartilhamentos  
214           25      0.0              200.0

**QUESTÃO 05:** Qual o post mais salvo?


> **Resposta:** O post mais salvo é SemanaPreFuvest_Imagem_Feed_Instagram_060121, o qual foi salvo 81 vezes.



In [254]:
# Procurar o post que mais foi salvo

mais_sav = df.sort_values("Saves").tail(1)
mais_sav

Nome do Post        Data  \
214  SemanaPreFuvest_Imagem_Feed_Instagram_060121  01/06/2021   

            Criativo Posicionamento Plataforma Formato  Conteúdo  Alcance  \
214  SemanaPreFuvest           Feed  Instagram  Imagem       0.0   6058.0   

     Impressões  Saves  Cliques em Link  3-Sec-View  Visualizações 100%  \
214      8849.0   81.0              0.0         0.0                 0.0   

     Comentários  Reações  Compartilhamentos  
214           25      0.0              200.0

**QUESTÃO 06:** Qual o total de *saves* que nosso *Instagram* possui?

> **Resposta**: Possui um total de 266 *saves*.

In [255]:
# Somar a quantidade de saves de cada postagem

saves = df["Saves"]
total_saves = sum(saves)
total_saves

266.0

**QUESTÃO 07:** Em quais posts alcançamos mais usuários?

> **Resposta:** Os posts que mais alcançaram usuários, em ordem decrescente, são: DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagram_140621 (com alcance de 6600 usuários), SemanaPreFuvest_Imagem_Feed_Instagram_060121 (com alcance de 6058 usuários) e ReelsAlissa_Vídeo_Reels_Instagram_211121 (com alcance de 3113 usuários).

In [256]:
# Ordenar, em ordem decrescente, as três primeiras postagens
# com o maior valor de alcance

maior_alc = df.sort_values("Alcance", ascending = False).head(3)
maior_alc

Nome do Post        Data  \
147  DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagra...  14/06/2021   
214       SemanaPreFuvest_Imagem_Feed_Instagram_060121  01/06/2021   
199           ReelsAlissa_Vídeo_Reels_Instagram_211121  21/11/2021   

                    Criativo Posicionamento Plataforma Formato  Conteúdo  \
147  DiaMundiaDoadorDeSangue        Stories  Instagram   Vídeo       0.0   
214          SemanaPreFuvest           Feed  Instagram  Imagem       0.0   
199              ReelsAlissa          Reels  Instagram   Vídeo       0.0   

     Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
147   6600.0        67.0    0.0              0.0         0.0   
214   6058.0      8849.0   81.0              0.0         0.0   
199   3113.0      3374.0   11.0              0.0         0.0   

     Visualizações 100%  Comentários  Reações  Compartilhamentos  
147                 0.0            1      0.0                0.0  
214                 0.0           25      0.0              200.0  
199                 0.0           22    112.0                0.0

**QUESTÃO 08:** Em quantos dias tivemos posts?
> **Resposta:** Houve posts em 91 dias.

In [258]:
# Desconsiderar as datas que aparecem repetidamente,
# utilizando um método que remove as repetições

df_sem_rep = df.drop_duplicates(subset = ["Data"])
len(df_sem_rep)

91

**QUESTÃO 09:** Quantos posts fizemos?
> **Resposta:** Foram feitos 230 posts.

In [259]:
# Calcular o número de colunas, já que
# cada coluna corresponde a um post distinto.

len(df)

230

**QUESTÃO 10:** Qual o total de usuários alcançados no nosso *Instagram*?

> **Resposta:** O máximo de usuários alcançados no perfil do *Instagram* é de 6600 usuários.


In [263]:
# Calcular o máximo de usuários já alcançados no Instagram

max_alc = df.sort_values("Alcance").tail(1)
max_alc

Nome do Post        Data  \
147  DiaMundiaDoadorDeSangue_Vídeo_Stories_Instagra...  14/06/2021   

                    Criativo Posicionamento Plataforma Formato  Conteúdo  \
147  DiaMundiaDoadorDeSangue        Stories  Instagram   Vídeo       0.0   

     Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
147   6600.0        67.0    0.0              0.0         0.0   

     Visualizações 100%  Comentários  Reações  Compartilhamentos  
147                 0.0            1      0.0                0.0

**QUESTÃO 11:** Qual o formato de post com mais compartilhamentos?
> **Resposta:** O formato de post mais compartilhado é o tipo Imagem, com um total de 471 compartilhamentos.

In [260]:
# Dividir os posts de acordo com o seu formato
# e calcular a soma de compartilhamentos para cada grupo.

df.groupby("Formato").sum(["Compartilhamentos"])

Conteúdo  Alcance  Impressões  Saves  Cliques em Link  3-Sec-View  \
Formato                                                                      
Imagem        0.0  33077.0     46074.0  247.0          45542.0         0.0   
Vídeo         0.0  15639.0      9394.0   19.0              0.0         0.0   

         Visualizações 100%  Comentários   Reações  Compartilhamentos  
Formato                                                                
Imagem                  0.0          746    4785.0              471.0  
Vídeo                   0.0           49  100157.0               14.0

**QUESTÃO 12:** O post sobre a federação da IME Jr teve melhor performance no Instagram ou no LinkedIn?
> **Resposta**: De forma muito significativa, o post sobre a federação da IME Jr. obteve melhor performance no *Instagram* em relação ao *LinkedIn*. A partir do cálculo da diferença dos valores de alcance e engajamento, é possível perceber que, em quase todos os critérios de avaliação da performance, a plataforma *Instagram* apresentou um desempenho extremamente superior ao *LinkedIn*. Pode-se dizer que o *Instagram* mostrou ser, em média, 136 vezes mais eficiente do que o *LinkedIn* para essa postagem.

In [279]:
# Separar os posts de acordo com a plataforma e avaliar 
# as performances individuais, baseando-se no maior alcance
# e maior engajamento dos usuários.

df[df["Criativo"] == "IMEJrFederada"]

Nome do Post        Data       Criativo  \
171  IMEJrFederada_Imagem_Feed_Instagram_090921  09/09/2021  IMEJrFederada   
172   IMEJrFederada_Imagem_Feed_LinkedIn_090921  09/09/2021  IMEJrFederada   

    Posicionamento Plataforma Formato  Conteúdo  Alcance  Impressões  Saves  \
171           Feed  Instagram  Imagem       0.0    599.0       669.0    2.0   
172           Feed   LinkedIn  Imagem       0.0      0.0       340.0    0.0   

     Cliques em Link  3-Sec-View  Visualizações 100%  Comentários  Reações  \
171              0.0         0.0                 0.0            8     53.0   
172             14.0         0.0                 0.0            2     17.0   

     Compartilhamentos  
171                0.0  
172                1.0

In [293]:
# Calcular as diferenças de métricas entre os posts

alcance = df.iloc[171]["Alcance"]-df.iloc[172]["Alcance"]
impressoes = df.iloc[171]["Impressões"]-df.iloc[172]["Impressões"]
saves = df.iloc[171]["Saves"]-df.iloc[172]["Saves"]
cliques = df.iloc[171]["Cliques em Link"]-df.iloc[172]["Cliques em Link"]
comentarios = df.iloc[171]["Comentários"]-df.iloc[172]["Comentários"]
reacoes = df.iloc[171]["Reações"]-df.iloc[172]["Reações"]
compartilhamentos = df.iloc[171]["Compartilhamentos"]-df.iloc[172]["Compartilhamentos"]

print(f"A diferença de usuários alcançados pelo Instagram em relação ao LinkedIn é de {alcance} usuários.")
print(f"A diferença de impressões no Instagram em relação ao LinkedIn é de {impressoes} impressões.")
print(f"A diferença de posts salvos no Instagram em relação ao LinkedIn é de {saves} posts.")
print(f"A diferença de cliques em link através do Instagram em relação ao LinkedIn é de {cliques} cliques.")
print(f"A diferença de comentários realizados no Instagram em relação ao LinkedIn é de {comentarios} comentarios.")
print(f"A diferença de reações no Instagram em relação ao LinkedIn é de {reacoes} reações.")
print(f"A diferença de compartilhamentos feitos no Instagram em relação ao LinkedIn é de {compartilhamentos} compartilhamentos.")

A diferença de usuários alcançados pelo Instagram em relação ao LinkedIn é de 599.0 usuários.
A diferença de impressões no Instagram em relação ao LinkedIn é de 329.0 impressões.
A diferença de posts salvos no Instagram em relação ao LinkedIn é de 2.0 posts.
A diferença de cliques em link através do Instagram em relação ao LinkedIn é de -14.0 cliques.
A diferença de comentários realizados no Instagram em relação ao LinkedIn é de 6 comentarios.
A diferença de reações no Instagram em relação ao LinkedIn é de 36.0 reações.
A diferença de compartilhamentos feitos no Instagram em relação ao LinkedIn é de -1.0 compartilhamentos.


In [297]:
media = (alcance + impressoes + saves + cliques + comentarios + reacoes + compartilhamentos)/7
print(media)

136.71428571428572


**QUESTÃO 13:** Qual a média de curtidas no *Instagram*? E no *LinkedIn*?
> **Resposta:** A média de curtidas no *Instagram* é de 496,42 curtidas e, no *LinkedIn*, é de 94,913 curtidas, aproximadamente.

In [261]:
# Dividir os grupos "Instagram" e "LinkedIn", 
# para, assim, calcular a média de curtidas para cada. 

insta = df[df.Plataforma == "Instagram"]
link = df[df.Plataforma == "LinkedIn"]

mi = insta["Reações"].mean()
ml = link["Reações"].mean()

instagram = pd.Series({'Plataforma': 'Instagram', 'Média de curtidas': mi})
linkedin = pd.Series({'Plataforma': 'LinkedIn', 'Média de curtidas': ml})
tabela_curt = pd.DataFrame([instagram, linkedin])
tabela_curt

Plataforma  Média de curtidas
0  Instagram         496.420290
1   LinkedIn          94.913043

In [ ]:
# QUESTÃO 14 : Há algum post com dados esquisitos? Quais as suas hipóteses?
# RESPOSTA   : 199

In [ ]:
# QUESTÃO 15 :  Qual mês teve mais reações?
# RESPOSTA   : 


In [ ]:
# QUESTÃO 16 : Quantos e quais posts tiveram mais do que 15 comentários?
# RESPOSTA   : 8 posts obtiveram mais de 15 comentários. São eles:
#              DiaDaFamília_Imagem_Feed_Instagram_150521	
#              DiaDoProfessor_Imagem_Feed_Instagram_151021	
#              DiaMusicistas_Imagem_Reels_Instagram_221121	
#              FotoConfraternização_Imagem_Feed_Instagram_211221	
#              NovaGestão_Imagem_Feed_Instagram_141221	
#              ReelsAlissa_Vídeo_Reels_Instagram_211121
#              SemanaPreFuvest_Imagem_Feed_Instagram_060121	
#              SorteioValePresente_Imagem_Feed_Instagram_291021

m_quinze = df[df.Comentários > 15]
print(f'{len(m_quinze)} posts obtiveram mais de 15 comentários.')
m_quinze

In [ ]:
# QUESTÃO 17 : Quantos posts do feed não possuem comentários? E quantos stories?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 18 : Quantos posts no feed temos no Instagram desde 29/10/2021?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 19 : Quantas vezes cada post foi visto por usuário? Qual o post mais visto por usuário?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 20 : Qual o post com a melhor performance de reações?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 21 : Qual(is) dia(s) da semana temos mais reações?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 22 : Qual o total de impressões que os posts com criativo DiaDoDesafio tiveram?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 23 : Quantos likes teve nosso post da Isabelle Tamarozi? Ele foi um bom post? Por quê?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 24 : Qual o nome do nosso melhor story? Por quê?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 25 : Qual o nome do nosso melhor post? Por quê?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 26 : Posts que têm mais curtidas têm mais comentários também?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 27 : Crie uma coluna de nota para cada post. O cálculo da nota fica a seu critério.
# RESPOSTA   : 

In [ ]:
# QUESTÃO 28 : Tivemos uma melhora ou piora de performance após a divulgação 
#              da nova identidade visual? 
#              Quais hipóteses você levanta pra isso?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 29 : Perdemos muito engajamento com posts sequenciais?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 30 : Qual post deveríamos repetir?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 31 : Qual formato a gente deve priorizar?
# RESPOSTA   : 

In [ ]:
# QUESTÃO 32 : Compare nossa performance com outra EJ.
# RESPOSTA   : 

In [ ]:
# QUESTÃO 33 : Como tomador(a) de decisão, quais seriam suas estratégias pro time de Marketing?
# RESPOSTA   : 